cou/cli.py

```diff
@@ -291,4 +292,6 @@ def entrypoint() -> None:
     finally:
         if args.command == "upgrade":
             loop.run_until_complete(run_post_upgrade_sanity_check(args))
+        if log_file:
```

Comment:
I think this conditional can be removed; `log_file` will always be populated.

Reply:
No, it's defensive programming. We shouldn't trust the line before `log_file = get_log_file()`.

Reply:
Oh sorry my bad, I didn't realise that was in the try block.  Can we move the logging setup to before the try block?  Ideally we'd only need `loop.run_until_complete(_run_command(args))` in the try block.

But that can be out of scope for this PR if you'd prefer.

Reply:
Actually, if it crashes before `log_file = ...`, `log_file` won't even be declared, so the `if log_file` will crash.  We'd need to at least declare log_file before the `try`.

Reply:
Since get_log_file return always Path and never None, I do not see reason for using `if log_file`. 
Also, I do not think that this is good approach start indicator at one function and end it here, where it can "stop" some other "start".

Reply:
> Can we move the logging setup to before the try block? Ideally we'd only need loop.run_until_complete(_run_command(args)) in the try block.

It's out of the scope.

Reply:
> Since get_log_file return always Path and never None, I do not see reason for using if log_file.

What if the line, `log_level = get_log_level(...)` failed? You can't suppose the lines before we assign the value is always no exception.

Reply:
> Actually, if it crashes before log_file = ..., log_file won't even be declared, so the if log_file will crash. We'd need to at least set log_file to something falsey before the try

Yes, this one is missing. I will update this part.

Reply:
> Also, I do not think that this is good approach start indicator at one function and end it here, where it can "stop" some other "start".

Maybe we can just use `print`

Reply:
> What if the line, log_level = get_log_level(...) failed? You can't suppose the lines before we assign the value is always no exception.

I see that you add `log_file: Optional[Path] = None`, so now it makes more sense.